File: backend/src/test/java/lan/dk/podcastserver/service/properties/BackupTest.java
package lan.dk.podcastserver.service.properties;

import org.junit.Before;
import org.junit.Test;

import java.nio.file.Paths;

import static lan.dk.podcastserver.assertion.Assertions.assertThat;

/**
 * Created by kevin on 13/04/2016 for Podcast Server
 */
public class BackupTest {

    private Backup backup;

    @Before
    public void beforeEach() {
        backup = new Backup();
    }

    @Test
    public void should_have_default_value() {
        /* Given */
        /* When */
        /* Then */
        assertThat(backup)
                .hasLocation(null)
                .hasCron("0 0 4 * * *")
                .hasBinary(false);
    }

    @Test
    public void should_have_specified_values() {
        /* Given */
        /* When */
        backup
            .setBinary(true)
            .setCron("cron")
            .setLocation(Paths.get("/tmp"));

        /* Then */
        assertThat(backup)
                .hasBinary(true)
                .hasCron("cron")
                .hasLocation(Paths.get("/tmp"))
        ;
    }

}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/service/properties/BackupTest.kt
package com.github.davinkevin.podcastserver.service.properties

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.nio.file.Paths

/**
 * Created by kevin on 13/04/2016 for Podcast Server
 */
class BackupTest {

    @Test
    fun should_have_default_value() {
        /* Given */
        /* When */
        val backup = Backup()
        /* Then */

        assertThat(backup.location).isEqualTo(Paths.get("/tmp"))
        assertThat(backup.cron).isEqualTo("0 0 4 * * *")
        assertThat(backup.binary).isEqualTo(false)
    }

    @Test
    fun should_have_specified_values() {
        /* Given */
        /* When */
        val backup = Backup(
                binary = true,
                cron = "cron",
                location = Paths.get("/foo")
        )
        /* Then */
        assertThat(backup.binary).isEqualTo(true)
        assertThat(backup.cron).isEqualTo("cron")
        assertThat(backup.location).isEqualTo(Paths.get("/foo"))
    }

}
